crates/pathfinder/src/sync/state_updates.rs

```diff
@@ -128,34 +130,65 @@ impl ProcessStage for VerifyCommitment {
     }
 }
 
-pub struct Store {
-    db: pathfinder_storage::Connection,
-    current_block: BlockNumber,
+pub struct UpdateStarknetState {
+    pub storage: pathfinder_storage::Storage,
+    pub connection: pathfinder_storage::Connection,
+    pub current_block: BlockNumber,
+    pub verify_tree_hashes: bool,
 }
 
-impl Store {
-    pub fn new(db: pathfinder_storage::Connection, start: BlockNumber) -> Self {
-        Self {
-            db,
-            current_block: start,
-        }
-    }
-}
-
-impl ProcessStage for Store {
-    const NAME: &'static str = "StateDiff::Persist";
+impl ProcessStage for UpdateStarknetState {
     type Input = StateUpdateData;
     type Output = BlockNumber;
 
+    const NAME: &'static str = "StateDiff::UpdateStarknetState";
+
     fn map(&mut self, state_update: Self::Input) -> Result<Self::Output, SyncError2> {
         let mut db = self
-            .db
+            .connection
             .transaction()
             .context("Creating database transaction")?;
 
         let tail = self.current_block;
 
-        db.insert_state_update_data(self.current_block, &state_update)
+        let header = db
+            .block_header(tail.into())
+            .context("Querying block header")?
+            .context("Block header not found")?;
+        let parent_state_commitment = match self.current_block.parent() {
+            Some(parent) => db
+                .state_commitment(parent.into())
+                .context("Querying parent block header")?
+                .context("Parent block header not found")?,
+            None => StateCommitment::default(),
+        };
```

Comment:
I don't think `update_starknet_state` uses the parent state commitment. Maybe we're passing the wrong type in here for that function -- it actually only uses certain parts of the state update: contract_updates, system_contract_updates and declared_sierra_classes.